iceoryx_posh/test/moduletests/test_popo_trigger.cpp

```diff
@@ -425,3 +455,146 @@ TEST_F(Trigger_test, TriggerIsNotLogicalEqualWhenInvalid)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    sut.invalidate();
+
+    EXPECT_FALSE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, InvalidEventBasedTriggerIsNotLogicalEqualToSameEventOriginAndNonEmptyHasTriggeredCallback)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    sut.invalidate();
+
+    EXPECT_FALSE(sut.isLogicalEqualTo(&m_triggerClass, {m_triggerClass, &TriggerClass::hasTriggered}));
+}
+
+TEST_F(Trigger_test, InvalidEventBasedTriggerIsNotLogicalEqualToDifferentEventOriginAndEmptyHasTriggeredCallback)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    sut.invalidate();
+    TriggerClass anotherTriggerClass;
+
+    EXPECT_FALSE(sut.isLogicalEqualTo(&anotherTriggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, InvalidEventBasedTriggerIsNotLogicalEqualToDifferentEventOriginAndNonEmptyHasTriggeredCallback)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    sut.invalidate();
+    TriggerClass anotherTriggerClass;
+
+    EXPECT_FALSE(sut.isLogicalEqualTo(&anotherTriggerClass, {m_triggerClass, &TriggerClass::hasTriggered}));
+}
+
+TEST_F(Trigger_test, ValidEventBasedTriggerUpdateOriginWorks)
+{
+    TriggerClass anotherTriggerClass;
+    Trigger sut = createValidEventBasedTrigger();
+    sut.updateOrigin(anotherTriggerClass);
+
+    EXPECT_TRUE(sut.isLogicalEqualTo(&anotherTriggerClass, cxx::ConstMethodCallback<bool>()));
+    EXPECT_FALSE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, InvalidEventBasedTriggerUpdateOriginDoesNotWork)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    sut.invalidate();
+    TriggerClass anotherTriggerClass;
+    sut.updateOrigin(anotherTriggerClass);
+
+    EXPECT_FALSE(sut.isLogicalEqualTo(&anotherTriggerClass, cxx::ConstMethodCallback<bool>()));
+    EXPECT_FALSE(sut.isLogicalEqualTo(&m_triggerClass, cxx::ConstMethodCallback<bool>()));
+}
+
+TEST_F(Trigger_test, EventBasedTriggerWithEmptyResetCallCallsErrorHandlerAndIsInvalid)
+{
+    const uint64_t eventId = 0U;
+    const uint64_t uniqueTriggerId = 0U;
+
+    bool hasTerminated = false;
+    iox::Error errorType = iox::Error::kNO_ERROR;
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            hasTerminated = true;
+            errorType = error;
+        });
+
+    Trigger sut(EventBasedTrigger,
+                &m_triggerClass,
+                cxx::MethodCallback<void, uint64_t>(),
+                eventId,
+                TriggerClass::callback,
+                uniqueTriggerId);
+
+    EXPECT_TRUE(hasTerminated);
+    EXPECT_THAT(errorType, Eq(iox::Error::kPOPO__TRIGGER_INVALID_RESET_CALLBACK));
+    EXPECT_FALSE(sut.isValid());
+    EXPECT_FALSE(static_cast<bool>(sut));
+}
```

Comment:
Maybe I overlooked it: Is there a test where you try to construct an EventBasedTrigger with an empty reset callback?

Reply:
Yes, its the test case: `EventBasedTriggerWithEmptyResetCallCallsErrorHandlerAndIsInvalid`

Reply:
Oh, it was right above :D